peer.go

```diff
@@ -454,6 +424,130 @@ func (p *peer) loadActiveChannels(chans []*channeldb.OpenChannel) error {
 	return nil
 }
 
+// addLink creates and adds a new link from the specified channel.
+func (p *peer) addLink(chanPoint *wire.OutPoint,
+	lnChan *lnwallet.LightningChannel,
+	forwardingPolicy *htlcswitch.ForwardingPolicy,
+	blockEpoch *chainntnfs.BlockEpochEvent,
+	chainEvents *contractcourt.ChainEventSubscription,
+	currentHeight int32, syncStates bool) error {
+
+	// onChannelFailure will be called by the link in case the channel
+	// fails for some reason.
+	onChannelFailure := func(chanID lnwire.ChannelID,
+		shortChanID lnwire.ShortChannelID,
+		linkErr htlcswitch.LinkFailureError) {
+
+		// The link has notified us about a failure. We launch a go
+		// routine to stop the link, disconnect the peer and optionally
+		// force close the channel. We must launch a goroutine since we
+		// must let OnChannelFailure return in order for the link to
+		// completely stop in the call to RemoveLink.
+		p.wg.Add(1)
+		go func() {
+			defer p.wg.Done()
+
+			// We begin by removing the link from the switch, such
+			// that it won't be attempted used for any more
+			// updates.
+			err := p.server.htlcSwitch.RemoveLink(chanID)
```

Comment:
All links for this peer will cleaned up by the `peerTerminationWatcher` upon detecting a disconnect. If the link has already broken its event loop, why is this needed?

Reply:
It was meant to have the effect of making sure it is not attempted to be used for routing, since we know at this point that the link won't be useful. 

Not sure how much of an effect it will have though, but my assumption was that it might take a while before the `peerTerminationWatcher` removes it?

Reply:
Also note that the call to `ForceClose` below will also call `RemoveLink`, but that is only in the few force close cases, ofc.

Reply:
Ahh I see, yeah sounds like it can't hurt!